Cargo.toml

```diff
@@ -45,6 +45,7 @@ tracing = "0.1"
 sql-builder = "3"
 tempfile = { version = "3.19" }
 minijinja = "2.9.0"
+metrics = "0.21.1"
```

Comment:
this change is the issue. please don't try and update metrics alongside adding this feature. the metrics update is, unfortunately, far more complex than that.